<launch>
    <arg name="gui" default="true" />
    <arg name="test" default="false" />

    <!-- send the bigman robot XML to param server -->
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find qb_legs_description)/model/qb_legs_robot.urdf.xacro'" />

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
        <param name="use_gui" value="$(arg gui)"/>
        <rosparam param="source_list">[simple_chain_ik/joint_states]</rosparam>
    </node>

    <!-- start robot state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen" />


    <!-- load parameters on the server -->
    <rosparam file="$(find simple_chain_ik)/config/qb_legs_params.yaml" command="load"/>


    <group unless="$(arg test)">
        <node name="simple_chain_ik_node" pkg="simple_chain_ik" type="simple_chain_ik_node" output="screen" />
    </group>
    <group if="$(arg test)">
        <node name="simple_chain_ik_test" pkg="simple_chain_ik" type="simple_chain_ik_test" output="screen"/>
    </group>

</launch>
